samples/csharp/getting-started/Ranking_Web/README.md

```diff
@@ -0,0 +1,155 @@
+
+## Dataset
+The training and testing data used by this sample is based on a public [dataset provided by Microsoft](https://www.microsoft.com/en-us/research/project/mslr/) originally provided Microsoft Bing. The dataset is released under a [CC-by 4.0](https://creativecommons.org/licenses/by/4.0/) license.  
+
+@article{DBLP:journals/corr/QinL13,
+  author    = {Tao Qin and 
+               Tie{-}Yan Liu},
+  title     = {Introducing {LETOR} 4.0 Datasets},
+  journal   = {CoRR},
+  volume    = {abs/1306.2597},
+  year      = {2013},
+  url       = {https://arxiv.org/abs/1306.2597},
+  timestamp = {Mon, 01 Jul 2013 20:31:25 +0200},
+  biburl    = {https://dblp.uni-trier.de/rec/bib/journals/corr/QinL13},
+  bibsource = {dblp computer science bibliography, https://dblp.org}
+}
+
+The following description is provided for this dataset:
+
+    The datasets are machine learning data, in which queries and urls are represented by IDs. The datasets consist of feature vectors extracted from query-url pairs along with relevance judgment labels:
+
+    * The relevance judgments are obtained from a retired labeling set of a commercial web search engine (Microsoft Bing), which take 5 values from 0 (irrelevant) to 4 (perfectly relevant).
+
+    * The features are basically extracted by us (e.g. Microsoft), and are those widely used in the research community.
+
+    In the data files, each row corresponds to a query-url pair. The first column is relevance label of the pair, the second column is query id, and the following columns are features. The larger value the relevance label has, the more relevant the query-url pair is. A query-url pair is represented by a 136-dimensional feature vector.
+
+## ML Task - Ranking
+As previously mentioned, this sample uses the LightGBM LambdaRank algorithm which is applied using a supervised learning technique known as [**Learning to Rank**](https://en.wikipedia.org/wiki/Learning_to_rank).  This technique requires that train\test datasets contain groups of data instances that are each labeled with their relevance score (e.g. relevance judgment label).  The label is a numerical\ordinal value, such as {0, 1, 2, 3, 4}.  The process for labeling these data instances with their relevance scores can be done manually by subject matter experts.  Or, the labels can be determined using other metrics, such as the number of clicks on a given search result. 
+
+It is expected that the dataset will have many more "Bad" relevance scores than "Perfect".  This helps to avoid converting a ranked list directly into equally sized bins of {0, 1, 2, 3, 4}.  The relevance scores are also reused so that you will have many items **per group** that are labeled 0, which means the result is "Bad".  And, only one or a few labeled 4, which means that the result is "Perfect". 
```

Comment:
Might be nice to show a bar / pie chart of label values from this MSLR dataset. This would exemplify the expected distribution of labels for ranking.

The breakdown for this demo's MSLR dataset:
* Label 0 -- 624,263
* Label 1 -- 386,280
* Label 2 -- 159,451
* Label 3 -- 21,317
* Label 4 -- 8,881

This would be nice in graph form, but just the numeric table form is slightly useful, as the user could calculate that there are 70x as many 0 "Bad", then 4 "Perfect" labels in the dataset.

Reply:
I added the breakdown in text format - #resolved